### PR TITLE
Trivial implementation of TS

### PR DESCRIPTION
**Pull request by sladymon**
This is an initial trivial implementation of part of the TS (translation server) using capitalization as a stub for translation. See the README for usage.

### CODE REVIEW

File: ts/po_localizer.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+    for f in os.listdir(in_path):
+        if f.endswith('.po'):
+            localize_po_file(in_path, out_path, f, lang)
+
+
+def localize_po_file(in_path, out_path, po_file, lang):
+    """Parses the given po file to find all msgid(untranslated string)
+        and add lang translation in corresponding msgstr(translated string)
+
```
Comment (liquidsteves) [inline ts/po_localizer.py]: The msgid/msgstr thing is confusing if you don't know that's a PO file format thing. Maybe link to https://www.gnu.org/software/gettext/manual/html_node/PO-Files.html or something?

File: ts/README.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Usage
```
Comment (liquidsteves) [inline ts/README.md]: I haven't heard people saying "ts" as much as they say "tms", so not sure it will be obvious what is stands for. Should we spell it out as "translation_services" or something

File: ts/po_localizer.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+                        wf.write(line)
+                        wf.write('"' + loc_text + '"\n')
+                    text = ""  # reset text
+                    msgstr_started = False
+                else:
+                    # Write original line
+                    wf.write(line)
+
+
+def translate_text(text, lang):
```
Comment (liquidsteves) [inline ts/po_localizer.py]: We will want to keep this "uppercase string" translation as an option long term, because it makes testing logic for localize_po_file() much easier. It's nice to be able to test your own local code without calling out to a potentially slow and unpredictable external api.

File: ts/README.md (hunk 2 above)
Comment (sladymon) [reply ts/README.md]: Sure! I was just using the same naming convention as in the Serge configuration file, but we can easily change this.

File: ts/po_localizer.py (hunk 3 above)
Comment (sladymon) [reply ts/po_localizer.py]: That makes sense. I'll add in another arg for the program that allows the user to specify which translation to use.

**Comment by sladymon** [comment]
Refactored code is now up on this branch (although there is a merge conflict with .gitignore). I'll rebase after tests have been added  (and possibly after we have a docker image).

File: translation_service/src/translators.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        # Send list of texts to Google Translate
+        translation_list = self.google_client.translate(texts_list, target_language=target_lang_iso)
+
+        # Grab the translated texts from the list of dictionaries returned by Google
+        localized_texts = [translation.get('translatedText') for translation in translation_list]
+
+        return localized_texts
+
+
+class TranslatorFactory:
```
Comment (liquidsteves) [inline translation_service/src/translators.py:72]: Yay abstraction!

File: translation_service/src/translators.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+    @staticmethod
+    def get_translator(translation_api, google_key_path=None):
```
Comment (liquidsteves) [inline translation_service/src/translators.py]: Make the default explicit in the method signature by doing translation_api="caps".

Put the validation logic for these two arguments in this method

File: translation_service/src/project_localizer.py [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+    if not os.path.isdir(output_dir):
+        sys.exit("ERROR: Output directory does not exist.")
+
+    # Check that the translation API is supported
+    if translation_api not in SUPPORTED_TRANSLATION_APIS:
+        sys.exit("ERROR: Translation API is not supported. Supported APIs include: {}.".format(
+            SUPPORTED_TRANSLATION_APIS))
+
+    if translation_api == "google" and google_key_path is None:
+        sys.exit("ERROR: To use the Google Translate API, you must include the google_key_path.")
```
Comment (liquidsteves) [inline translation_service/src/project_localizer.py]: Put this validation logic (and the definition of SUPPORTED_TRANSLATION_APIS) in the translators.py instead.

File: translation_service/test/test_po_localizer.py [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+        write_po_localized_file
+            expected in: localized_po_filepath, po_lines, localized_texts
+            expected out: localized .po file which has the localized strs (golden file)
+
+        # Call write_po_localized_file
+        # check that the file is written to the correct place
+        # check that the file matches golden file (with golden input for po_lines, localized_texts)
+        # check anything else?
+        # check any edge cases?
+        """
```
Comment (liquidsteves) [inline translation_service/test/test_po_localizer.py:93]: Simply having comments like these is actually really valuable for documentation of how the code is supposed to work, even if the test isn't implemented yet. Likewise, having the code refactored so that you'll be able to add test cases when you want to is also very valuable, since you've created some very clean APIs dividing up the logic.

**Review by liquidsteves — APPROVED** [review]
Thanks for doing all this refactoring and testability work! This is awesome!!!

File: translation_service/src/translators.py (hunk 5 above)
Comment (sladymon) [reply translation_service/src/translators.py]: Will do!

File: translation_service/src/project_localizer.py (hunk 6 above)
Comment (sladymon) [reply translation_service/src/project_localizer.py]: Will do!
